### PR TITLE
Add `getDefaultCommitmentRequestTransformer` helper

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR creates a new `getDefaultCommitmentRequestTransformer` function that extracts all the logic related to transforming parameters in order to provide a default commitment when appropriate.

### CODE REVIEW

**Comment by lorisleiva** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3159?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3159** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3159?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3158** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3158?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3150** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3150?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3149** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3149?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3148** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3148?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3147** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3147?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3146** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3145** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/rpc-transformers/src/params-transformer.ts [hunk 1]
```diff
@@ -28,23 +28,11 @@ export function getDefaultRequestTransformerForSolanaRpc(config?: RequestTransfo
...
-                methodName,
-                params: applyDefaultCommitment({
-                    commitmentPropertyName: methodName === 'sendTransaction' ? 'preflightCommitment' : 'commitment',
-                    optionsObjectPositionInParams,
-                    overrideCommitment: defaultCommitment,
-                    params: params as unknown[],
-                }),
+            getDefaultCommitmentRequestTransformer({
+                defaultCommitment,
+                optionsObjectPositionByMethod: OPTIONS_OBJECT_POSITION_BY_METHOD,
```
Comment (buffalojoec) [inline packages/rpc-transformers/src/params-transformer.ts:35]: Sweet! So someone could also use `pipe(...)` to create a custom request transformer that builds on top of `getDefaultCommitmentRequestTransformer`.

**Review by buffalojoec — APPROVED** [review]

**Review by steveluscher — APPROVED** [review]
Added via Giphy

**Comment by lorisleiva** [comment]
### Merge activity

* **Sep 1, 2:06 PM EDT**: @lorisleiva started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3159).

**Pull request merged**